store/file/ods.go

```diff
@@ -97,15 +89,40 @@ func CreateODSFile(
 	}, nil
 }
 
-func writeODSFile(w io.Writer, h *headerV0, eds *rsmt2d.ExtendedDataSquare) error {
+func writeODSFile(w io.Writer, eds *rsmt2d.ExtendedDataSquare, axisRoots *share.AxisRoots) (*headerV0, error) {
+	// write header
+	h := &headerV0{
+		fileVersion: fileV0,
+		fileType:    ods,
+		shareSize:   share.Size,
+		squareSize:  uint16(eds.Width()),
+		datahash:    axisRoots.Hash(),
+	}
 	err := writeHeader(w, h)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("writing header: %w", err)
+	}
+
+	// write axisRoots
+	err = writeAxisRoots(w, axisRoots)
+	if err != nil {
+		return nil, fmt.Errorf("writing axis roots: %w", err)
 	}
 
 	for _, shr := range eds.FlattenedODS() {
 		if _, err := w.Write(shr); err != nil {
-			return err
+			return nil, fmt.Errorf("writing shares: %w", err)
+		}
+	}
+	return h, nil
+}
+
+func writeAxisRoots(w io.Writer, roots *share.AxisRoots) error {
+	for _, roots := range [][][]byte{roots.RowRoots, roots.ColumnRoots} {
+		for _, root := range roots {
+			if _, err := w.Write(root); err != nil {
```

Comment:
Do we buffer writes? Feels like a lot of independent writes

Reply:
I've tried buffering write of axisRoots there is barely any benefit. But this idea made me also try to buffering of shares when writing quadrants. It improved Put time by 3x. Will add in another PR.